README.md

```diff
@@ -10,27 +10,68 @@ One of the main objectives of this project is to supply a good starting point fo
+More precisely, the main components of Redux are State, Action, and Reducer.
+In Gong the composables communicate actions to the viewModels so they can manage and emit the state back to view.
+This ensure state can only be updated in a predictable way.
+Then, inside the model, the reducer is called with a proper action and the latest state and forward its result as an output value of the model.
+Reducer is a function that takes the previous state and action and creates a new state, and in Gong this role is played by the function ‘processAction’, located in viewModels.
+
+To continue with the insight of the project, let's see how this is done.
+With the shared flow, actions are broadcast to an unknown number (zero or more) of subscribers.
+In the absence of a subscriber, any posted action is immediately dropped. It is a design pattern to use for actions that must be processed immediately or not at all.
+The viewModel handles each action in the ‘processAction’ method. Whenever an action is added to the "contract", it also has to be added here.
+So all actions can be managed from the same place.
+With the channel, each event is delivered to a single subscriber.
+An attempt to post an event without subscribers will suspend as soon as the channel buffer becomes full, waiting for a subscriber to appear. Posted events are never dropped by default.
+Then to handle this Ui effects "oneSHotEvents" are used. Because Channels are hot and it is not necessary to show side effect again when orientation changed or UI become visible again.
+Finally, for handling UiState, StateFlow is used. StateFlow is just like LiveData but have initial value. So a state is always present.
+It's also a kind of SharedFlow. It's always expected to receive last view state when UI become visible.
 
 <p align="center">
-  <img src="/images/layers.png">
+  <img src="/images/GongMainV2Arq.png">
 </p>
 
+
+Now, as a way to give you an overview of the other layers and how the interaction with the presentation layer is done, let's review it's components.
+
+To the presentation layer, the UseCases are the ones who resolve each invocation from the ViewModels, and they both interact using coroutine library.
+A UseCase is a reusable component that might be used from different ViewModels.
+Same goes for Repositories, a repository can stand on its own without the vm and be re-used from other places.
+All these classes exist with a clear goal and purpose. The logic is split sensibly.
+Repositories have the function of communication between Domain Layer and Data Layer.
+More precisely, with coroutines help, they have to implement the necessary logic so they can call Remote and Local sources methods.
+
+At the end of the chain, as mentioned, Data Layer is found. It is responsible for persisting and obtaining all the data required for the model using different sources.
+The remote sources are the ones who manage interaction with the different endpoints.
+The local sources manage data base logic.
```

Comment:
I'd add that the repositories use the store library to combine the different sources